module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTabTest.php

```diff
@@ -0,0 +1,194 @@
+                            "class" => "class01",
+                            "label" => "label01",
+                            "permission" => "permission01",
+                            "selected" => false,
+                            "url" => "http://newurl1"
+                            ],
+                            ["id" => "A02",
+                            "class" => "class02",
+                            "label" => "label02",
+                            "permission" => "permission02",
+                            "selected" => false,
+                            "url" => "http://newurl2"
+                            ],],
+                            null
+                            ],
+            ];
+    }
+
+    /**
+     * Test getting the active tab.
+     *
+     * @param array $tabEnv tabConfig
+     * @param array $expectedResult expected result from getActiveTab
+     *
+     * @return void
+     *
+     * @dataProvider tabConfigProvider
+     */
+    public function testGetActiveTab(array $tabEnv, $expectedResult): void
+    {
+        $obj = new SwitchTab();
+        $this->assertSame($expectedResult, $obj->getActiveTab($tabEnv));
+    }
+
+    /**
+     * Data provider for testGetActiveTab.
+     *
+     * @return array
+     */
+    public function inactiveTabConfigProvider(): array
+    {
+        return ['Test1' => [[["id" => "A01",
+                            "class" => "class01",
+                            "label" => "label01",
+                            "permission" => "permission01",
+                            "selected" => true,
+                            "url" => "http://newurl1"
+                            ],
+                            ["id" => "A02",
+                            "class" => "class02",
+                            "label" => "label02",
+                            "permission" => "permission02",
+                            "selected" => false,
+                            "url" => "http://newurl2"
+                            ],],
+                            [["id" => "A02",
+                            "class" => "class02",
+                            "label" => "label02",
+                            "permission" => "permission02",
+                            "selected" => false,
+                            "url" => "http://newurl2"
+                ]]],
+            ];
+    }
+
+    /**
+     * Test getting the active tab.
+     *
+     * @param array $tabEnv tabConfig
+     * @param array $expectedResult expected result from getInactiveTabs
+     *
+     * @return void
+     *
+     * @dataProvider inactiveTabConfigProvider
+     */
+    public function testGetInactiveTab(array $tabEnv, array $expectedResult): void
+    {
+        $obj = new SwitchTab();
+        $this->assertSame($expectedResult, $obj->getInactiveTabs($tabEnv));
+    }
+
+    /**
+     * Test init method.
+     *
+     * @return void
+     */
+    public function testInit(): void
```

Comment:
I'm not sure if we need tests for init, setConfig, or process, since none of them currently do anything. It's hard to come up with meaningful assertions for these. My concern is that, if logic is added to these methods in the future, it's possible that these tests could provide coverage without actually asserting anything useful, which turn out to be a problem... but if you think they're of value, I'm not insisting that we remove them. :-)